lib/client.js

```diff
@@ -32,10 +32,13 @@ function defaultId () {
 function sendPacket (client, packet, cb) {
   try {
     var buf = mqttPacket.generate(packet);
-    if (!client.stream.write(buf) && cb) {
-      client.stream.once('drain', cb);
-    } else if (cb) {
+
+    client.emit('packetsend', packet);
```

Comment:
why are you emitting `'packetsend'` here?
Can you send another PR for `'packetsend'`  and `'packetreceive'`, so it's easier to discuss?